src/WebJobs.Script/JsonSerializerOptionsProvider.cs

```diff
@@ -19,6 +19,12 @@ public static class JsonSerializerOptionsProvider
         /// </summary>
         public static readonly JsonSerializerOptions Options = CreateJsonOptions();
 
+        /// <summary>
+        /// Shared Json serializer with the following setting:
+        /// - PropertyNameCaseInsensitive: true.
+        /// </summary>
+        public static readonly JsonSerializerOptions CaseInsensitiveJsonSerializerOptions = CreateCaseInsensitiveJsonOptions();
```

Comment:
There is a `JsonSerializerOptions.Web`, which has ProperyNameCaseInsensitive= true, PropertyNamingPolicy= camelCase, and NumberHandling = AllowReadingFromString.

This seems like it might work for your use case?